src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/NonArrivalReasonEntityFactory.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.factory
+
+import io.github.bluegroundltd.kfactory.Factory
+import io.github.bluegroundltd.kfactory.Yielded
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.NonArrivalReasonEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.randomStringMultiCaseWithNumbers
+import java.util.UUID
+
+class NonArrivalReasonEntityFactory : Factory<NonArrivalReasonEntity> {
+  private var id: Yielded<UUID> = { UUID.randomUUID() }
+  private var name: Yielded<String> = { randomStringMultiCaseWithNumbers(6) }
+  private var isActive: Yielded<Boolean> = { true }
+
+  fun withId(id: UUID) = apply {
+    this.id = { id }
+  }
+
+  fun withName(name: String) = apply {
+    this.name = { name }
+  }
+
+  fun withIsActive(isActive: Boolean) = apply {
+    this.isActive = { isActive }
+  }
+
+  override fun produce(): NonArrivalReasonEntity = NonArrivalReasonEntity(
+    id = this.id(),
+    name = this.name(),
+    isActive = this.isActive()
+  )
```

Comment:
Just thinking about these factories - is there a way we can provide defaults for some of the values? Would save a lot of the repetition in tests where we don't actually care about the values.

Reply:
Yeah so the default values are set when we declare the variables at the top of the class
```
  private var id: Yielded<UUID> = { UUID.randomUUID() }
  private var name: Yielded<String> = { randomStringMultiCaseWithNumbers(6) }
  private var isActive: Yielded<Boolean> = { true }
```

So you don't need to specify any of them in this case if you don't want to, e.g. just do:

```
NonArrivalEntityFactory().produce()
```

which will give you a NonArrival with random id/name and isActive of true